frame/support/src/weights/weight_v2.rs

```diff
@@ -0,0 +1,301 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use core::ops::{Add, Mul, Sub};
+use sp_runtime::{
+	traits::{CheckedAdd, One, Zero},
+	RuntimeDebug,
+};
+
+use super::*;
+
+#[derive(
+	Encode, Decode, MaxEncodedLen, TypeInfo, Eq, PartialEq, Copy, Clone, RuntimeDebug, Default,
+)]
+pub struct WeightV2 {
+	/// The weight of computational time used.
+	pub time: TimeWeight,
+	/// The weight of the storage bandwidth used.
+	pub bandwidth: StorageWeight,
+}
```

Comment:
```suggestion
pub struct WeightV2 {
	/// The weight of computational time used.
	pub computation: ComputationWeight,
	/// The weight of the storage bandwidth used.
	pub bandwith: BandwithWeight,
}
```

Time is not a good name because computation and transmission of the witness both take time.